zad6/config/routes.rb

```diff
@@ -1,3 +1,22 @@
 Rails.application.routes.draw do
+  resources :products
+
+
+  # Customers controllers setup
+
+  devise_for :customers, controllers: {
+    registrations: 'registrations'
+  }
+
+  # Customers paths setup
+  devise_scope :customer do
```

Comment:
`devise_scope`, wg [dokumentacji](https://www.rubydoc.info/github/plataformatec/devise/ActionDispatch%2FRouting%2FMapper%3Adevise_scope) służy głównie do precyzowania customowych ścieżek związanych z logowaniem/rejestracją. Tutaj użycie tej metody wydaje mi się zbędne. Zupełnie wystarczyłoby:
```ruby
resources :customers, only: [:index, :show] do
  resources :products, only: :index
end
```
To załatwiłoby wszystkie interesujące nas sytuację i zmusiło do odpowiedniego nazwania kontrolerów 😉 Oczywiście trzeba by było jeszcze pomyśleć co zrobić z faktem, że wyżej mamy `resources :products` i do akcji `products#index` kierują dwie ścieżki - `/products` i `/customers/:customer_id/products`. W tym przypadku można pomyśleć o użyciu [czegoś takiego](https://guides.rubyonrails.org/v3.2/routing.html#controller-namespaces-and-routing):
```ruby
  resources :customers, only: [:index, :show] do
    scope module: :customer do
      resources :products, only: :index
    end
  end
```
To generuje taki route:
```
customer_products GET    /customers/:customer_id/products    customer/products#index
```
Co pozwala na skorzystanie z modułu i stworzenie oddzielnego kontrolera dla produktów customera - `Customer::ProductsController`.
Możesz spróbować takiej czystszej struktury routesów i kontrolerów jeśli zechcesz podłubać jeszcze przy tym mini-projekciku. Tak czy inaczej - w obecnej wersji też jakoś to działa 👍